R/tm_g_pp_patient_timeline.R

```diff
@@ -471,11 +470,16 @@ tm_g_pp_patient_timeline <- function(label,
     plot_width[1],
     lower = plot_width[2], upper = plot_width[3], null.ok = TRUE, .var.name = "plot_width"
   )
-  assertthat::assert_that(!xor(is.null(aetime_start), is.null(aetime_end)))
-  assertthat::assert_that(!xor(is.null(dstime_start), is.null(dstime_end)))
-  assertthat::assert_that(!xor(is.null(aerelday_start), is.null(aerelday_end)))
-  assertthat::assert_that(!xor(is.null(dsrelday_start), is.null(dsrelday_end)))
-  assertthat::assert_that(
+  if (is.null(aetime_start) + is.null(aetime_end) == 1)
+    stop("Both 'aetime_start' and 'aetime_end' need to be provided or both need to be empty.")
```

Comment:
I suggest making the error message similar to `checkmate` 

```suggestion
  stop(
    "Assertion on `aetime_start` and `aetime_end` failed:",
    " Both `aetime_start` and `aetime_end` needs to be provided or both need to be `NULL`"
  )
```

Or with a fancy function and reuse it.

```r
xor_error_string <- function(x, y) {
  paste(
    "Assertion on `", x, "` and `", y, "` failed:",
    "Both `", x, "` and `", y, "` needs to be provided or both need to be `NULL`."
  ) 
}
```

Reply:
I like the function!

Reply:
Used the function and decided to go back to XOR lol : P c0e24e06ee93f18b80debdc5556858a1808ea58e

Reply:
The error messaging function makes total sense to me! It is clear and informative for users where xor is not